x-pack/plugins/cloud_integrations/cloud_experiments/server/plugin.ts

```diff
@@ -0,0 +1,135 @@
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+import { get, has } from 'lodash';
+import LaunchDarkly, { type LDClient, type LDUser } from 'launchdarkly-node-server-sdk';
+import type { LogMeta } from '@kbn/logging';
+import type { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import { registerUsageCollector } from './usage';
+import type { CloudExperimentsConfigType } from './config';
+import type {
+  CloudExperimentsFeatureFlagNames,
+  CloudExperimentsMetric,
+  CloudExperimentsPluginSetup,
+  CloudExperimentsPluginStart,
+} from '../common';
+import { FEATURE_FLAG_NAMES, METRIC_NAMES } from '../common/constants';
+
+interface CloudExperimentsPluginSetupDeps {
+  usageCollection?: UsageCollectionSetup;
+}
+
+export class CloudExperimentsPlugin
+  implements
+    Plugin<
+      CloudExperimentsPluginSetup,
+      CloudExperimentsPluginStart,
+      CloudExperimentsPluginSetupDeps
+    >
+{
+  private readonly logger: Logger;
+  private readonly launchDarklyClient?: LDClient;
+  private readonly flagOverrides?: Record<string, unknown>;
+  private launchDarklyUser: LDUser | undefined;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    const config = initializerContext.config.get<CloudExperimentsConfigType>();
+    if (config.flag_overrides) {
+      this.flagOverrides = config.flag_overrides;
+    }
+    const ldConfig = config.launch_darkly; // If the plugin is enabled and no flag_overrides are provided (dev mode only), launch_darkly must exist
+    if (!ldConfig && !initializerContext.env.mode.dev) {
+      // If the plugin is enabled, and it's in prod mode, launch_darkly must exist
+      // (config-schema should enforce it, but just in case).
+      throw new Error(
+        'xpack.cloud_integrations.experiments.launch_darkly configuration should exist'
+      );
+    }
+    if (ldConfig) {
+      this.launchDarklyClient = LaunchDarkly.init(ldConfig.sdk_key, {
+        application: { id: `kibana-server`, version: initializerContext.env.packageInfo.version },
+        logger: LaunchDarkly.basicLogger({ level: ldConfig.client_log_level }),
+        // For some reason, the stream API does not work in Kibana. `.waitForInitialization()` hangs forever (doesn't throw, neither logs any errors).
+        // Using polling for now until we resolve that issue.
+        // Relevant issue: https://github.com/launchdarkly/node-server-sdk/issues/132
+        stream: false,
+      });
+      this.launchDarklyClient.waitForInitialization().then(
+        () => this.logger.debug('LaunchDarkly is initialized!'),
+        (err) => this.logger.warn(`Error initializing LaunchDarkly: ${err}`)
+      );
+    }
+  }
+
+  public setup(
+    core: CoreSetup,
+    deps: CloudExperimentsPluginSetupDeps
+  ): CloudExperimentsPluginSetup {
+    this.logger.debug('cloudExperiments: Setup');
```

Comment:
The setup/start logger statements are redundant, we already write those from core: https://github.com/elastic/kibana/blob/9b75efdd1e1ec7cdd39af85fb34f1e5f70b915d3/src/core/server/plugins/plugins_system.ts#L102

Reply:
I'll remove them. To be fair, they were automatically added by the plugin generator, so I thought they were the recommendation

Reply:
> they were automatically added by the plugin generator

ooh, that's good to know. yeah we should probably remove those from the generator eventually.

Reply:
Removed these entries :)